### PR TITLE
stp_global fixes (includes interface properties)

### PR DESCRIPTION
**Pull request by mikewiebe**
- Add fretta_camden n8k support.
- Refactor interface.rb `check_switchport_disabled` method to check for any switchport state.
- Added checks for switchport mode `trunk` and `access` in stp interface properties.
- stp_global test cleanup/refactoring.

**stp_global tests**

| Platform | Image | Result |
| --- | --- | --- |
| n3k | I2 | :white_check_mark: |
| n5k | N1 | :white_check_mark: |
| n6k | N1 | :white_check_mark: |
| n7k | D1 | :white_check_mark: |
| n8k | fretta_camden | :white_check_mark: |
| n9k | I2 | :white_check_mark: |
| n9k | I4 (dublin plus) | :white_check_mark: |
| xr | n/a | n/a |

**interface (interface, svi, switchport) tests**

| Platform | Image | Result |
| --- | --- | --- |
| n9k | I2 | :white_check_mark: same as baseline |
| n9k | I4 (dublin plus) | :white_check_mark: same as baseline |
| xr | 6.0.0.21I | :white_check_mark: same as baseline |

### CODE REVIEW

File: lib/cisco_node_utils/interface.rb [hunk 1]
```diff
@@ -1094,10 +1103,14 @@ def switchport_vtp_feature?
       config_get('vtp', 'feature')
     end
 
-    def check_switchport_disabled
-      return if switchport_mode == :disabled || switchport_mode.nil?
```
Comment (mikewiebe) [inline lib/cisco_node_utils/interface.rb]: I was hoping we did not need the `switchport_mode.nil?` bit but looks like ti won't play nicely with XR.  New diffs coming...

File: tests/test_stp_global.rb [hunk 2]
```diff
@@ -24,36 +24,26 @@ class TestStpGlobal < CiscoTestCase
...
+    config("default interface #{interfaces[0]}") unless @@clean
     @@clean = true # rubocop:disable Style/ClassVars
+    @intf.switchport_enable
   end
 
   def teardown
     config 'no spanning-tree mode'
-    config 'system bridge-domain none' if n7k_platform?
+    config 'system bridge-domain none' if /N7/ =~ node.product_id
+    @intf.switchport_enable
```
Comment (robert-w-gries) [inline tests/test_stp_global.rb]: Why does the teardown method need to enable switchport mode?

**Comment by robert-w-gries** [comment]
:+1:

LGTM after comment is addressed

File: tests/test_stp_global.rb (hunk 2 above)
Comment (mikewiebe) [inline tests/test_stp_global.rb]: It doesn't.  Removed.

**Pull request merged**